technical-reference/reference-technical-articles/prime.md

```diff
@@ -0,0 +1,206 @@
+- 10% of the CAKE total income should be allocated to Prime → 100 CAKE (following the protocol tokenomics)
+- We should convert 100 CAKE to USDC, USDT, BTC and ETH, because in Prime the rewards are defined in these tokens
+- The conversion should follow the previous table:
+    - 19.24 CAKE should be converted to USDC
+    - 41.94 CAKE should be converted to USDT
+    - 17.70 CAKE should be converted to BTC
+    - 21.13 CAKE should be converted to ETH
+
+Interest reserves (part of the protocol income) from Isolated pools and core pool markets are sent to the PSR ([Protocol Share Reserve](https://github.com/VenusProtocol/protocol-reserve/blob/main/contracts/ProtocolReserve/ProtocolShareReserve.sol)) contract. Based on configuration a certain percentage of income from all markers is reserved for Prime token holders. The interest reserves will be sent to the PSR periodically (currently every 24 hours, but this can be changed by the community via [VIP](https://app.venus.io/governance)).
+
+The PSR has a function `releaseFunds` that needs to be invoked to release the funds to the destination contracts. We have [SingleTokenConverter](https://github.com/VenusProtocol/protocol-reserve/contracts/TokenConverter/SingleTokenConverter.sol) contracts which receive income from the PSR and convert them to Prime supported reward tokens. 
+
+Each `SingleTokenConverter` sends funds to `PrimeLiquidityProvider` contract which releases the funds to `Prime` contract. Distribution speeds for each of the reward token is configured in the `PrimeLiquidityProvider` contract and based on those speeds `Prime` distirbutes rewards. 
+
+When a user claims their rewards and if the `Prime` contract doesn’t have enough funds then we trigger the release of funds from `PrimeLiquidityProvider` to `Prime` contract in the same transaction i.e., in the `claimInterest` function.
+
+The following diagram shows the integration of the `SingleTokenConverter` contracts with the Prime contracts:
+
+<figure><img src="../../.gitbook/assets/prime_token_converter.png" alt="Integration of the SingleTokenConverter contracts with the Prime contracts"><figcaption></figcaption></figure>
+
+More information about Income collection and distribution can be found [here](../../whats-new/automatic-income-allocation.md).
+
+## Update cap multipliers and alpha
+
+Market multipliers and alpha can be updated at anytime and need to be propagated to all users. Changes will be gradually applied to users as they borrow/supply assets and their individual scores are recalculated. This strategy has limitations because the scores will be wrong in aggregate.
+
+To mitigate this issue, Venus will supply a script that will use the permission-less function `updateScores` to update the scores of all users. This script won’t pause the market or `Prime` contract. Scores need to be updated in multiple transactions because it will run out of gas trying to update all scores in 1 transaction.
```

Comment:
```suggestion
To mitigate this issue, Venus will supply a script that will use the permission-less function `updateScores` to update the scores of all users. This script won’t pause the market or `Prime` contracts. Scores need to be updated in multiple transactions because it will run out of gas trying to update all scores in 1 transaction.
```